### PR TITLE
chore: remove deprecated 'new-window' event

### PR DESCRIPTION
**Pull request by miniak**
#### Description of Change
It has been deprecated since Electron 13. Warning was added in Electron 20 (#34528).

#### Checklist
- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes
Notes: The deprecated `new-window` event has been removed.

### CODE REVIEW

**Review by nornagon — CHANGES_REQUESTED** [review]
i'm worried about removing this without any machinery to explode when people try to do things the old way. we used to recommend `new-window` listeners for security reasons, and i bet a lot of apps still use it. removing it without an error/warning seems dangerous.

i suggest a 2-version deprecate/delete cycle:
1. in version `X`, add a warning when an app registers a `new-window` listener.
2. in version `X+1`, remove the event and throw an error when the app registers a `new-window` listener.

**Comment by miniak** [comment]
@zcbenz can you please help addressing this?
https://github.com/electron/electron/blob/16cd22ff6a4dcccc3b8cc3f959b83b0d3f1ac68f/lib/browser/api/web-contents.ts#L690-L694
https://github.com/electron/electron/blob/16cd22ff6a4dcccc3b8cc3f959b83b0d3f1ac68f/lib/browser/guest-window-manager.ts#L219-L222

**Comment by miniak** [comment]
@nornagon can we still add the warning in Electron 20 to be able to land the removal in Electron 21?

**Comment by nornagon** [comment]
@miniak I'm not necessarily against that, but it's been deprecated for over a year now so I'm not sure waiting one more version to delete it is that bad? Is there any rush?

**Comment by samuelmaddock** [comment]
Adding a deprecation log and waiting seems like the correct approach. We may not be considering a certain use case and users can voice their concern prior to removal if they notice the deprecation.

If we can afford it, maybe waiting 2 versions would be worthwhile since not everyone upgrades to the latest version.

**Comment by miniak** [comment]
@samuelmaddock the question was mainly whether to do warning in E21 and then remove in E22, or already warn in E20 and remote in E21. We can also warn in E20 and remove in E22.

**Comment by samuelmaddock** [comment]
> We can also warn in E20 and remove in E22.

This sounds best to me to give folks upgrading ample time.

**Review by jkleinsc — APPROVED** [review]
API LGTM

**Review by nornagon — APPROVED** [review]

**Comment by nornagon** [comment]
API LGTM

**Review by zcbenz — APPROVED** [review]
API LGTM

File: docs/breaking-changes.md [hunk 1]
```diff
@@ -12,6 +12,24 @@ This document uses the following convention to categorize breaking changes:
 * **Deprecated:** An API was marked as deprecated. The API will continue to function, but will emit a deprecation warning, and will be removed in a future release.
 * **Removed:** An API or feature was removed, and is no longer supported by Electron.
 
+## Planned Breaking API Changes (22.0)
```
Comment (stefansundin) [inline docs/breaking-changes.md:15]: This should say `21.0` and not `22.0` right?

**Pull request merged**